feat(metrics): add value pool, RPC, and peer health metrics

gustavovalverde opened the pull request:
## Motivation

Operators lack visibility into Zebra's internal state. This PR adds Prometheus metrics for:
- Value pool balances (transparent, sprout, sapling, orchard, deferred)
- RPC request rates, latencies, and errors
- Peer handshake performance and failure reasons

Closes #10162 
Closes #10163
Closes #10164
Closes #10167

<img width="2574" height="1608" alt="image" src="https://github.com/user-attachments/assets/0322008e-7127-4fd2-a66c-cc9f0f63e25f" />

## Solution

### Value Pool Metrics (`zebra-state`)

New metrics in `zebra_db/metrics.rs`:
- `state.finalized.value_pool.{transparent,sprout,sapling,orchard,deferred}` - Pool balances in zatoshis
- `state.finalized.chain_supply.total` - Total chain supply

These are for monitoring/observability. Zebra already enforces ZIP-209 internally by rejecting blocks that would cause negative pools.

### RPC Metrics (`zebra-rpc`)

New middleware in `server/rpc_metrics.rs`:
- `rpc.requests.total{method,status}` - Request count by method and status
- `rpc.request.duration_seconds{method}` - Latency histogram
- `rpc.active_requests` - Concurrent requests gauge
- `rpc.errors.total{method,code}` - Error count by method and code

### Peer Health Metrics (`zebra-network`)

Added to `peer/handshake.rs`:
- `zcash.net.peer.handshake.duration_seconds{result}` - Handshake latency histogram
- `zcash.net.peer.handshake.failures.total{reason}` - Failure count by reason

### Dashboards & Alerts

- `docker/observability/grafana/dashboards/value_pools.json` - Value pool visualization
- `docker/observability/grafana/dashboards/rpc_metrics.json` - RPC performance dashboard
- `docker/observability/prometheus/rules/zebra_alerts.yml` - Alert rules for critical conditions

### Tests

Tested with the local observability stack (`docker compose -f docker/docker-compose.observability.yml up`):
- Verified all metrics appear at `/metrics` endpoint
- Confirmed dashboards render correctly in Grafana
- Validated alert rules load in Prometheus

### PR Checklist

- [x] The PR name is suitable for the release notes.
- [x] The PR follows the contribution guidelines.
- [x] The library crate changelogs are up to date.
- [x] The solution is tested.
- [x] The documentation is up to date.

gustavovalverde committed:
feat(metrics): add value pool, RPC, and peer health metrics

Add Prometheus metrics for operational visibility:

- Value pool metrics: transparent, sprout, sapling, orchard, deferred
  balances and total chain supply (zebra-state)
- RPC metrics middleware: request counts, latencies, active requests,
  and errors per method (zebra-rpc)
- Peer handshake metrics: duration histograms and failure tracking
  by reason (zebra-network)

Also includes:
- Grafana dashboards for value pools and RPC monitoring
- Prometheus alert rules for critical conditions
- Updated CHANGELOG

Closes #10162, closes #10163, closes #10164, closes #10167

gustavovalverde committed:
chore: remove wrongly commited files to `.gitignore`

gustavovalverde committed:
chore: cargo fmt

gustavovalverde committed:
chore: clippy fix

upbqdn commented on CHANGELOG.md:
```diff
@@ -13,6 +13,10 @@ and this project adheres to [Semantic Versioning](https://semver.org).
 - OpenTelemetry tracing support behind the `opentelemetry` feature flag ([#10174](https://github.com/ZcashFoundation/zebra/pull/10174))
 - RPC tracing middleware with `SPAN_KIND_SERVER` spans for Jaeger Service Performance Monitoring ([#10174](https://github.com/ZcashFoundation/zebra/pull/10174))
 - Added Docker Compose observability stack with Jaeger, Prometheus, Grafana, and AlertManager for local development and testing
```
```suggestion
- Docker Compose observability stack with Jaeger, Prometheus, Grafana, and AlertManager for local development and testing ([#10174](https://github.com/ZcashFoundation/zebra/pull/10174))
```

upbqdn commented on zebrad/src/components/mempool/downloads.rs:
```diff
@@ -167,7 +167,7 @@ where
                         Option<oneshot::Sender<Result<(), BoxError>>>,
                     ),
-                    (TransactionDownloadVerifyError, UnminedTxId),
+                    Box<(TransactionDownloadVerifyError, UnminedTxId)>,
```
Why are we boxing this?

gustavovalverde committed:
docs: fix changelog entries for PR #10175

- Remove #10174 entries from [Unreleased] (already in 4.0.0)
- Remove redundant "Added" prefix from entries
- Link to PR #10175 instead of issues
- Add rpc_metrics module entry to zebra-rpc changelog

gustavovalverde committed:
Update CHANGELOG.md

Co-authored-by: Marek <mail@marek.onl>

gustavovalverde committed:
Update zebra-rpc/CHANGELOG.md

Co-authored-by: Marek <mail@marek.onl>
